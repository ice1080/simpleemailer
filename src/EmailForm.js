import React, { Component } from 'react';

export default class EmailForm extends Component {
  constructor(props) {
    super(props);

    this.state = {
      to: '',
      to_name: '',
      from: '',
      from_name: '',
      subject: '',
      body: '',
    };

    this.bindMethods.call(this);
  }

  bindMethods() {
    this.handleTextChange = this.handleTextChange.bind(this);
    this.submitForm = this.submitForm.bind(this);
  }

  clearForm() {
    this.setState({to: '', to_name: '', from: '', from_name: '', subject: '', body: ''});
  }

  handleTextChange(event) {
    var updatedState = {};
    updatedState[event.target.name] = event.target.value;
    this.setState(updatedState);
  }

  submitForm() {
    let data = this.state; // I wouldn't normally do this, but since there's nothing else in state I'm using this for simplicity

    this.props.onSubmit(data);
  }
  
  render() {
    return (
      <div>
        Send an email via the configured provider.
        All Fields Required!
        <br/>
        <br/>
        To Email Address: <input type={'text'} name={'to'} value={this.state.to} onChange={this.handleTextChange} />
        <br/>
        <br/>
        To Name: <input type={'text'} name={'to_name'} value={this.state.to_name} onChange={this.handleTextChange} />
        <br/>
        <br/>
        From Email Address: <input type={'text'} name={'from'} value={this.state.from} onChange={this.handleTextChange} />
        <br/>
        <br/>
        From Name: <input type={'text'} name={'from_name'} value={this.state.from_name} onChange={this.handleTextChange} />
        <br/>
        <br/>
        Subject: <input type={'text'} name={'subject'} value={this.state.subject} onChange={this.handleTextChange} />
        <br/>
        <br/>
        Body: <textarea rows={'8'} columns={'100'} name={'body'} value={this.state.body} onChange={this.handleTextChange} />
        <br/>
        <br/>
        <button onClick={this.submitForm} >Send Email</button>
      </div>
    );
  }
}
