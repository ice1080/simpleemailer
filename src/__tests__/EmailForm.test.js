import React from 'react';
import {shallow, mount} from 'enzyme';
import EmailForm from '../EmailForm';

describe ('EmailForm', () => {
  let wrapper;
  let onSubmit = jest.fn();

  beforeEach(() => {
    wrapper = shallow(<EmailForm onSubmit={onSubmit} />);
  });

  afterEach(() => {
    onSubmit.mockReset();
  });

  test('default state', () => {
    expect(wrapper.state()).toMatchSnapshot();
  });

  test('clearForm', () => {
    wrapper.setState({
      to: 'to1',
      to_name: 'to_name1',
      from: 'from1',
      from_name: 'from_name1',
      subject: 'subject1',
      body: 'body1',
    });
    wrapper.instance().clearForm();
    expect(wrapper.state()).toMatchSnapshot();
  });

  test('handleTextChange', () => {
    let event = {target: {
      name: 'to',
      value: 'newToValue',
    }};
    wrapper.instance().handleTextChange(event);
    expect(wrapper.state().to).toBe('newToValue');
  });

  test('submitForm', () => {
    wrapper.setState({
      to: 'to2',
      to_name: 'to_name2',
      from: 'from2',
      from_name: 'from_name2',
      subject: 'subject2',
      body: 'body2',      
    });

    expect(onSubmit).toHaveBeenCalledTimes(0);
    wrapper.instance().submitForm();
    expect(onSubmit).toHaveBeenCalledTimes(1);
    expect(onSubmit).toHaveBeenLastCalledWith({
      to: 'to2',
      to_name: 'to_name2',
      from: 'from2',
      from_name: 'from_name2',
      subject: 'subject2',
      body: 'body2',
    });
  });
});
