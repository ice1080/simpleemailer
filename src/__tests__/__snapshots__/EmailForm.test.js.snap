// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EmailForm clearForm 1`] = `
Object {
  "body": "",
  "from": "",
  "from_name": "",
  "subject": "",
  "to": "",
  "to_name": "",
}
`;

exports[`EmailForm default state 1`] = `
Object {
  "body": "",
  "from": "",
  "from_name": "",
  "subject": "",
  "to": "",
  "to_name": "",
}
`;
